base/deprecated.jl

```diff
@@ -1009,4 +1009,18 @@ export @vectorize_1arg, @vectorize_2arg
 @deprecate abs(M::SymTridiagonal) abs.(M)
 @deprecate abs(x::AbstractSparseVector) abs.(x)
 
+# #18218
+eval(Base.LinAlg, quote
+    function arithtype(T)
+        depwarn("arithtype is deprecated as it is no longer needed in Julia Base.",
```

Comment:
Yeah, you're right. @blegat Can you specify something more useful, like

"arithtype is now deprecated. If you were using inside a promote_op call consider doing `promote_op(matprod, Ts...)`. If you need instead its functionality, consider defining it locally."

Of course, it doesn't have to be that if you find a better way to express it.